src/repr/src/strconv.rs

```diff
@@ -615,9 +614,8 @@ pub fn parse_bytes(s: &str) -> Result<Vec<u8>, ParseError> {
     // [0]: https://www.postgresql.org/docs/current/datatype-binary.html#id-1.5.7.12.9
     // [1]: https://www.postgresql.org/docs/current/datatype-binary.html#id-1.5.7.12.10
     if let Some(remainder) = s.strip_prefix(r"\x") {
-        parse_bytes_hex(remainder).map_err(|e| {
-            ParseError::invalid_input_syntax("bytea", s).with_details(e.to_string_alt())
-        })
+        parse_bytes_hex(remainder)
+            .map_err(|e| ParseError::invalid_input_syntax("bytea", s).with_details(e.to_string()))
```

Comment:
This should include causes, right?

```suggestion
            .map_err(|e| ParseError::invalid_input_syntax("bytea", s).with_details(e.to_string_with_causes()))
```

Reply:
This is true, I do have an in-progress branch where I try and tackle `strconv.rs` as well.

Context: We have `DisplayExt::to_string_alt`, which was motivated by the observation that `anyhow` prints the error chain when you request the alternate selector.  We then have `ResultExt::err_to_string` and `ResultExt::map_err_to_string` which use `DisplayExt::to_string_alt`. I think this is a footgun, as we now know, because most other error types don't print the error chain when you request the alternate.

Back to `strconv.rs`: It's a heavy user of `err_to_string` and `map_err_to_string`, but only has a `Display` bound for it's error type `E`. I'll have to refactor that slightly to make it work with `Error` bounds on it's error type, and I think then I could also remove `DisplayExt` and have `ResultExt` be backed by `ErrorExt`. Wdyt?

Reply:
I think that's a great idea!